### ISSUE TITLE
Rename 'tlsstarted' to something that isn't a lie, and use more global constants for magic strings

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1983]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#1983|
|Type|task|
|Created|2006-08-03 15:35:25Z|

Before AMP is released, we should fix the known API maladies it has. Notably, 'tlsstarted' is a lie, because it's actually called when TLS negotiation is beginning, not when it has completed and we have a live TLS connection (hopefully, in the future, a real TLS-started event will be possible).

Also, all those magic strings that we rely on application developers to use should be mapped as global constants, to reduce errors.

Attachments:

* [remove-tls-started.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/1aa/1aa8a9a093cd1b8a820fd6e92c62fbc6ef375b62/4d4ad53cedc4d5802d3f1c77ee4c029e4ece85ed.patch) (4967 bytes) - added by radix on 2006-09-24 19:32:25Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1983 1983
type__task task
reporter__radix radix
priority__highest highest
milestone__Core_2_5 Core-2.5
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1154619325000000 1154619325000000
changetime__1160451582000000 1160451582000000
version__None None
owner__ 
cc__ed@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix set status to `new`|
|-|-|

Here is a patch that removes tls started crap. Please review someone.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @radix|
|-|-|

Looks good.  Merge.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_6"></a>|@radix set status to `closed`|
|-|-|

(In [18418]) remove the bogus "tls-started" stuff from AMP.

Review: exarkun
Author: radix
Fixes [#1983](https://github.com/twisted/twisted/issues/1983)

The tls_started callback was a lie. This change deletes it so as to
minimize any possible confusion.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
